examples/jupyter/demo010_advection.ipynb

```diff
@@ -0,0 +1,149 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Advection Example"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# dependencies\n",
+    "\n",
+    "using Pkg\n",
+    "Pkg.activate(\"./\")\n",
+    "Pkg.instantiate()\n",
+    "Pkg.develop(path=\"./../../../LFAToolkit.jl\")\n",
+    "using Plots\n",
+    "using LFAToolkit\n",
+    "using LinearAlgebra \n",
+    "using Polynomials"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# setup\n",
+    "p = 10\n",
+    "dimension = 1\n",
+    "collocate = false\n",
+    "mapping = haletrefethenstriptransformation(1.4)\n",
+    "\n",
+    "\n",
+    "mesh = []\n",
+    "if dimension == 1\n",
+    "   mesh = Mesh1D(1.0)\n",
+    "elseif dimension == 2\n",
+    "   mesh = Mesh2D(1.0, 1.0)\n",
+    "end\n",
+    "\n",
+    "# operator\n",
+    "advection = GalleryOperator(\"advection\", p + 1, p + 1, mesh)\n",
+    "mass = GalleryOperator(\"mass\", p + 1, p + 1, mesh, collocatedquadrature = collocate, mapping = mapping)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# compute full operator symbols\n",
```

Comment:
The 2D plotting doesn't seem to work
```
# compute full operator symbols
numbersteps = 250
maxeigenvalue = 0
θ_min = -π
θ_max = π
θ_step = 2π/(numbersteps-1)
θ_range = θ_min:θ_step:θ_max

# compute and plot smoothing factor
# -- 1D --
if dimension == 1
    # compute
    maxeigenvalues = zeros(numbersteps)
    mineigenvalues = zeros(numbersteps)
    for i in 1:numbersteps    
        θ = [θ_range[i]]
        A = computesymbols(advection, θ) * 2
        M = computesymbols(mass, θ)
        eigenvalues = hcat(sort(imag.(eigvals(-M\A)))...)'
        maxeigenvalues[i] = max(eigenvalues...)
        mineigenvalues[i] = min(eigenvalues...)
        maxeigenvalue = max(maxeigenvalue, maxeigenvalues[i])
    end

    # plot
    println("max eigenvalue: ", maximum(maxeigenvalues))
    xrange = θ_range/π
    plot(
        xrange,
        xlabel="θ/π",
        xtickfont=font(12, "Courier"),
        [maxeigenvalues, mineigenvalues, π*xrange],
        ytickfont=font(12, "Courier"),
        ylabel="spectral radius",
        linewidth=3,
        label=["Maximum λ" "Minimum λ" "θ"],
        title="Advection Operator Symbol Maximal Eigenvalues"
    )
    ymin = minimum(mineigenvalues)
    ylims!(minimum([0, ymin * 1.1]), maximum(maxeigenvalues) * 1.1)
# -- 2D --
elseif dimension == 2
    # compute
    maxeigenvalues = zeros(numbersteps, numbersteps)
    mineigenvalues = zeros(numbersteps, numbersteps)

    for i in 1:numbersteps, j in 1:numbersteps
        θ = [θ_range[i], θ_range[j]]
        A = computesymbols(advection, θ) * 2
        M = computesymbols(mass, θ)
        eigenvalues = hcat(sort(imag.(eigvals(-M\A)))...)'
        maxeigenvalues[i, j] = max(eigenvalues...)
        mineigenvalues[i, j] = max(eigenvalues...)
        maxeigenvalue = max(maxeigenvalue, maxeigenvalues[i, j])
    end

    # plot
    θ_max = θ_min + 2π
    θ_range = LinRange(θ_min, θ_max, numbersteps)
    println("max eigenvalue: ", maximum(maxeigenvalues))
    xrange = θ_range/π
    plot1 = heatmap(
        xrange,
        xlabel="θ/π",
        xtickfont=font(12, "Courier"),
        xrange,
        ylabel="θ/π",
        maxeigenvalues,
        ytickfont=font(12, "Courier"),
        title="Advection Operator Symbol Maximum Eigenvalues",
        transpose=true,
        aspect_ratio=:equal
    )
    xlims!(θ_min/π, θ_max/π)
    ylims!(θ_min/π, θ_max/π)
    plot2 = heatmap(
        xrange,
        xlabel="θ/π",
        xtickfont=font(12, "Courier"),
        xrange,
        ylabel="θ/π",
        mineigenvalues,
        ytickfont=font(12, "Courier"),
        title="Advection Operator Symbol Minimum Eigevalues",
        transpose=true,
        aspect_ratio=:equal
    )
    xlims!(θ_min/π, θ_max/π)
    ylims!(θ_min/π, θ_max/π)
    plot!(plot1, plot2, layout = (2, 1), size = (700, 1400))
end
```

Reply:
Pls, the 2D plot is not yet fixed in this recent push. I will further think and look into it.